### PR TITLE
[bitnami/redis] Add option to change port name in master service

### LINKED ISSUE #25228: [bitnami/redis] Ability to rename redis service port name
### Name and Version

bitnami/redis latest

### What is the problem this feature will solve?

Our internal kube cluster has an odd requirement, it requires all service `ports.name` to match the `ports.targetPort`, I've filed something internally to hopefully get this constraint fixed (but due to the complexity of fixing it, it may take a lot of time, or never be resolved).

```yaml
name: tcp-redis
targetPort: redis
```

https://github.com/bitnami/charts/blob/7359f94e1c61801f043b4515f697575830f814e1/bitnami/redis/templates/master/service.yaml#L47-L58

### What is the feature you are proposing to solve the problem?

It would be awesome, if `ports.name` or `ports.targetPort` name could also be configurable

### What alternatives have you considered?

I've tried using `.Values.master.service.extraPorts` to add a port with matching name

```yaml
service:
  extraPorts:
    - name: redis
      port: 6379
      protocol: TCP
      targetPort: redis
```

but that results with a duplicate key since the default config and the extraPorts config are using the same ports

```
spec.ports[1]: Duplicate value: core.ServicePort{Name:"", Protocol:"TCP", AppProtocol:(*string)(nil), Port:6379, TargetPort:intstr.IntOrString{Type:0, IntVal:0, StrVal:""}, NodePort:0}
```

### PR DESCRIPTION
**Pull request by cwgthornton**
### Description of the change

Lets you specify the port.name in the service for the redis master service.

### Benefits

If your kube instance requires the port's `name` to match the `targetPort` value, you can now use this chart.

### Possible drawbacks

Not aware of any

### Applicable issues

- fixes #25228

### Additional information

NA

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/redis/templates/master/service.yaml [hunk 1]
```diff
@@ -45,7 +45,11 @@ spec:
   externalIPs: {{- include "common.tplvalues.render" (dict "value" .Values.master.service.externalIPs "context" $) | nindent 4 }}
   {{- end }}
   ports:
+      {{- if .Values.master.service.portNames.redis }}
+    - name: {{ .Values.master.service.portNames.redis }}
+      {{- else }}
     - name: tcp-redis
+      {{- end }}
```
Comment (dgomezleon) [inline bitnami/redis/templates/master/service.yaml:52]:
```suggestion
    - name: {{ .Values.master.service.portNames.redis }}
```

File: bitnami/redis/Chart.yaml [hunk 2]
```diff
@@ -36,4 +36,4 @@ maintainers:
 name: redis
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/redis
-version: 19.2.0
+version: 19.2.1
```
Comment (dgomezleon) [inline bitnami/redis/Chart.yaml:39]: Bump a minor version when adding new values --> 19.3.0

**Review by dgomezleon — CHANGES_REQUESTED** [review]
Hi @cwgthornton 

Thanks for contributing. I added some minor suggestions to follow the same approach as RabbitMQ.

File: bitnami/redis/templates/master/service.yaml (hunk 1 above)
Comment (cwgthornton) [reply bitnami/redis/templates/master/service.yaml]: Done - also updated the default in values.yaml

**Review by dgomezleon — APPROVED** [review]
LGTM. Thanks

**Pull request merged**